packages/ember/tests/helpers/link_to_test/link_to_with_query_params_test.js

```diff
@@ -1,506 +1,577 @@
+    `);
+
+    this.visit('/').then(() => {
+      this.$('#the-link').click();
+      let indexController = this.applicationInstance.lookup('controller:index');
+
+      assert.deepEqual(indexController.getProperties('foo', 'bar'),
+        this.indexProperties,
+        'controller QP properties do not update'
+      );
+    });
+  }
+
+  [`@test doesn't update controller QP properties on current route when invoked (empty query-params obj)`](assert) {
+    this.addTemplate('index', `
+      {{#link-to 'index' (query-params) id='the-link'}}Index{{/link-to}}
+    `);
 
-      registry.unregister('router:main');
-      registry.register('router:main', Router);
+    this.visit('/').then(() => {
+      this.$('#the-link').click();
+      let indexController = this.applicationInstance.lookup('controller:index');
+
+      assert.deepEqual(indexController.getProperties('foo', 'bar'),
+        this.indexProperties,
+        'controller QP properties do not update'
+      );
     });
-  },
+  }
 
-  teardown: sharedTeardown
-});
+  [`@test doesn't update controller QP properties on current route when invoked (empty query-params obj, inferred route)`](assert) {
+    this.addTemplate('index', `
+      {{#link-to (query-params) id='the-link'}}Index{{/link-to}}
+    `);
 
-QUnit.test('doesn\'t update controller QP properties on current route when invoked', function() {
-  setTemplate('index', compile(`{{#link-to 'index' id='the-link'}}Index{{/link-to}}`));
-  bootApplication();
+    this.visit('/').then(() => {
+      this.$('#the-link').click();
+      let indexController = this.applicationInstance.lookup('controller:index');
 
-  run(jQuery('#the-link'), 'click');
-  let indexController = container.lookup('controller:index');
-  deepEqual(indexController.getProperties('foo', 'bar'), { foo: '123', bar: 'abc' }, 'controller QP properties not');
-});
+      assert.deepEqual(indexController.getProperties('foo', 'bar'),
+        this.indexProperties,
+        'controller QP properties do not update'
+      );
+    });
+  }
+
+  ['@test updates controller QP properties on current route when invoked'](assert) {
+    this.addTemplate('index', `
+      {{#link-to 'index' (query-params foo='456') id="the-link"}}
+        Index
+      {{/link-to}}
+    `);
+
+    this.visit('/').then(() => {
+      this.runTask(() => this.$('#the-link').click());
+      let indexController = this.applicationInstance.lookup('controller:index');
+
+      assert.deepEqual(indexController.getProperties('foo', 'bar'),
+        { foo: '456', bar: 'abc' },
+        'controller QP properties updated'
+      );
+    });
+  }
+
+  ['@test updates controller QP properties on current route when invoked (inferred route)'](assert) {
+    this.addTemplate('index', `
+      {{#link-to (query-params foo='456') id="the-link"}}
+        Index
+      {{/link-to}}
+    `);
+
+    this.visit('/').then(() => {
+      this.runTask(() => this.$('#the-link').click());
```

Comment:
@patricksrobertson `click` is already wrapped within a runTask here. So there is no need for it, i guess.

https://github.com/emberjs/ember.js/blob/master/packages/internal-test-helpers/lib/test-cases/abstract.js#L91